### PR TITLE
Update istio version for testing

### PR DESCRIPTION
**Pull request by ericvn**
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by ericvn** [comment]
Changes in istio test framework:
```
# istio.io/istio.io/pkg/test/istioio
pkg/test/istioio/builder.go:64:3: undefined: scopes.CI
pkg/test/istioio/builder.go:75:5: undefined: scopes.CI
pkg/test/istioio/script.go:77:2: undefined: scopes.CI
FAIL	istio.io/istio.io/tests/setup/profile_default [build failed]
FAIL	istio.io/istio.io/tests/setup/profile_demo [build failed]
FAIL	istio.io/istio.io/tests/setup/profile_none [build failed]
```

**Comment by ericvn** [comment]
The new feature code is failing our tests:
```
✔ Installation complete=== RUN   TestDocs
=== RUN   TestDocs/ops/configuration/mesh/app-health-check/test.sh
    TestDocs/ops/configuration/mesh/app-health-check/test.sh: test.go:211: Detected new test TestDocs/ops/configuration/mesh/app-health-check/test.sh in suite tests_setup_profile_default with no feature labels.  See istio/istio/pkg/test/framework/features/README.md
=== RUN   TestDocs/tasks/security/authentication/authn-policy/test.sh
    TestDocs/tasks/security/authentication/authn-policy/test.sh: test.go:211: Detected new test TestDocs/tasks/security/authentication/authn-policy/test.sh in suite tests_setup_profile_default with no feature labels.  See 
...
```

**Comment by frankbu** [comment]
> Will we need to whitelist all the tests, including new ones? Do we want to have the same sort of feature tracking that istio/istio has? Is a switch needed to turn off the checking?

These tests are all testing the same Istio feature (docs). Test coverage of the docs is tracked separately, so I don't see any point in using istio/istio feature tracking for these tests.

I think we need to turn off the feature tracking in the istioio test framework override.

File: pkg/test/framework/features/whitelist.txt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+tests_setup_profile_default, TestDocs/tasks/traffic-management/request-timeouts/test.sh
+tests_setup_profile_default, TestDocs/tasks/traffic-management/tcp-traffic-shifting/test.sh
+tests_setup_profile_default, TestDocs/tasks/traffic-management/traffic-shifting/test.sh
+tests_setup_profile_demo, TestDocs/examples/bookinfo/test.sh
+tests_setup_profile_demo, TestDocs/tasks/observability/logs/access-log/test.sh
+tests_setup_profile_demo, TestDocs/tasks/security/authentication/mtls-migration/test.sh
+tests_setup_profile_demo, TestDocs/tasks/traffic-management/egress/egress-gateway-tls-origination/mtls_test.sh
+tests_setup_profile_demo, TestDocs/tasks/traffic-management/egress/egress-gateway-tls-origination/tls_test.sh
+tests_setup_profile_none, TestDocs/tasks/security/cert-management/dns-cert/test.sh
+tests_setup_profile_none, TestDocs/tasks/security/cert-management/plugin-ca-cert/test.sh
```
Comment (frankbu) [inline pkg/test/framework/features/whitelist.txt:25]: We need to turn this off for all, not list the tests one by one.

Comment (ericvn) [reply pkg/test/framework/features/whitelist.txt:25]: Unfortunately, there isn't an option right now. Someone like @therealmitchconnors would have to add it to the framework.

Comment (ericvn) [reply pkg/test/framework/features/whitelist.txt:25]: Some option we can add when running to not output the message and the fatalf.

Comment (ericvn) [reply pkg/test/framework/features/whitelist.txt:25]: Another weird item I found was I had to have a slightly different suite name (adding a _work) when trying to run locally.

Comment (frankbu) [reply pkg/test/framework/features/whitelist.txt:25]: How about creating a new feature, `documentation`, and add a call to `Features` [after this line](https://github.com/istio/istio.io/blob/a7382bef813024cb3f9d1c95b898987578cf761a/pkg/test/istioio/framework.go#L193) to make them all implement that feature.

Comment (ericvn) [reply pkg/test/framework/features/whitelist.txt]: I think that works as well.

**Comment by ericvn** [comment]
Move from whitelist to using a feature. It's actually much simpler.

**Pull request merged**